### ISSUE TITLE
SWC Minify slowness

### ISSUE THREAD

**Issue by smitpatel1010** [vercel/next.js#42746]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
Binaries:
  Node: 16.15.1
  npm: 8.11.0
  Yarn: 1.22.19
Relevant packages:
  next: 12.3.2-canary.15
  eslint-config-next: 12.1.0
  react: 17.0.2
  react-dom: 17.0.2
```

### How are you deploying your application? (if relevant)

custom server somewhat similar to `https://github.com/vercel/next.js/blob/canary/examples/custom-server/server.ts` but with express

### Describe the Bug

swc minifiaction time is extremely high as compare to terser
<img width="1682" alt="Screenshot 2022-11-10 at 6 48 27 PM" src="https://user-images.githubusercontent.com/37947873/201102091-825e6b46-db26-4564-937a-afcc66b6c858.png">

How can I debug more in this?

FYI: `I verified that the issue exists in the latest Next.js canary release` I had to tick this checkbox to create an issue. but I have not tested it with latest canary because I can not use latest next with react 17

### Expected Behavior

swc minification should be fast.

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

-

### To Reproduce

-

**Comment by balazsorban44** [comment]
Hi, we will need a reproduction to investigate.

**Comment by smitpatel1010** [comment]
Hey @balazsorban44, uhh I am not sure about what is the actual issue here so that I can put that into separate repo to reproduce exact behaviour. Do you know how can I debug more on this?

**Comment by kdy1** [comment]
1. Install rust using https://rustup.rs

2. Clone swc ( git@github.com:swc-project/swc.git )

3. Run https://github.com/swc-project/swc/blob/6672c52af2d7c835a89c079e9649c171d9c433e0/crates/swc_ecma_minifier/scripts/next/add-test.sh from the app directory.
Once you do it, the inputs to the swc minifier will be coped to swc test directory.

4. From `./crates/swc_ecma_minifier`, run `cargo test --test compress --features concurrent --release`. I think it will then print error message like `foo test is running for 60 seconds`

5. Find the input file causing the slowness, and finally you can hand it over to me

**Comment by smitpatel1010** [comment]
Thanks @kdy1. I have a very large repo. so i tried removing few pages and ran the above. but there was no error message present besides failed tests due to output.js is not there.

I noticed one thing here that with pages removed it must have created the same chunk for existing pages. but surprisingly It did not take that much time.

in this screenshot https://github.com/vercel/next.js/issues/42746#issue-1443906706 minimum time was `221s`

```
terser-webpack-plugin-optimize 87s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 21s
├─ minify-js 20s
├─ minify-js 20s
├─ minify-js 20s
├─ minify-js 20s
├─ minify-js 20s
├─ minify-js 20s
```

**Comment by kdy1** [comment]
How much cpu cores do you have? I don't think the number of files matter, but the number of cpu core count matters.

**Comment by balazsorban44** [comment]
@smitpatel1010 if the code is private, could you add @kdy1 so he can have a closer look? That would accelerate the investigation of this issue. :pray:

**Comment by smitpatel1010** [comment]
Sorry, but can't give code access. and @kdy1 number cores must be more than 64. and also number of cores can effect total time. but time for individual file should not be dependent on number of cores right?

**Comment by kdy1** [comment]
No, swc minifier optimizes one file in parallel, and yeah I think the number of CPU core is the cause.
https://github.com/rayon-rs/rayon/issues/394

**Comment by smitpatel1010** [comment]
Ohh 🤔. So Is there any possible solution?

**Comment by kdy1** [comment]
Honestly, not sure - I didn't think I would hit the issue in real world because CPU with 64+ core is super expensive.

Can you try modification with `RAYON_NUM_THREADS=16`?

**Comment by smitpatel1010** [comment]
Yes. let me try

**Comment by smitpatel1010** [comment]
not working even with `RAYON_NUM_THREADS=16`.

**Comment by smitpatel1010** [comment]
Yes brother I understand that. and I am trying to make a reproduction. and I can't give code access because its not under my authority. and also Its not on github too.

**Comment by smitpatel1010** [comment]
@kdy1 I am getting one issue in swc transpilation and I want to reproduce it on swc playground. and its working with latest version but breaking with `1.2.249`. so How can I find which swc version is getting used in specific next version?

**Comment by smitpatel1010** [comment]
These 2 cases are breaking
* [Link 1](https://play.swc.rs/?version=1.2.203&code=H4sIAAAAAAAAA32PQQrCMBBF9z3FX6Yg7T5NI%2BIFBAuui86yyZBGUErv7qTWagXNYhje%2FLxJ6MY%2BRJy966UGaiPtmFFngGk2Vg3ofUeH4BmjxlH65s5kGpujthJSktaQknBiGDLBmI2n0DLTZVJCTQO84N537B25OGNgC5ZFfbKoBaZjvq%2FY1VgCacNQFMUkGFGuA6b8J8iXXr8f8MOYV8%2FvBYrX4D7%2BVwkeqwfy0RJxTwEAAA%3D%3D&config=H4sIAAAAAAAAA0WOSw7DIAxE7%2BJ1tu2CO%2FQQiDoRFT95HKkIcfcCSpXdaGb8PI0%2BcGQaFStgmQo1qf2SIa2F4cQXpY0Uw1I5uQ9t5WAdDcZjRCFnMJndBvBG0Se%2F1wlyORZh4I5sOsK%2F2Qco5vc5jbZ%2BLeCT%2Bs247jxeV3EN%2BAFJDkBftAAAAA%3D%3D)
* [Link 2](https://play.swc.rs/?version=1.2.203&code=H4sIAAAAAAAAA32PsQ7CIBCG9z7FP8JCd2kxxhdwc270xsIFMDEhvLuAtVoTZSCX77%2F74OjOzkdcnA3l9jRFOjBj7ACRENxMJ%2B8YWWI0hbXMIHWlxjJ19hMzXVsE0QK84NHN7CzZuGBgDy7CUC1ihfUM3yNmE5eG%2BkJSSjVBRr9tGPp%2FArnWu%2FcHfhilfq7nKd68%2FdhPF5z1Azzitb8zAQAA&config=H4sIAAAAAAAAA0WOSw7DIAxE7%2BJ1tu2CO%2FQQiDoRFT95HKkIcfcCSpXdaGb8PI0%2BcGQaFStgmQo1qf2SIa2F4cQXpY0Uw1I5uQ9t5WAdDcZjRCFnMJndBvBG0Se%2F1wlyORZh4I5sOsK%2F2Qco5vc5jbZ%2BLeCT%2Bs247jxeV3EN%2BAFJDkBftAAAAA%3D%3D)

**Comment by balazsorban44** [comment]
Please stop tagging maintainers until there is a reproduction provided. I am closing this for now. Open a new issue when there is a sufficient reproduction or the authority is given to have a look at the original code base. It's impractical for kdy1 to guess the problem.

SWC version can be seen here: https://github.com/vercel/next.js/blob/canary/packages/next-swc/crates/core/Cargo.toml#L45. As of writing, this corresponds to https://github.com/swc-project/swc/releases/tag/v1.3.19

If you are debugging via SWC only, the issue might be reported in https://github.com/swc-project/swc instead.
